src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/StatusBar/BarStyle.android.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using Android.App;
+using Android.OS;
+using Android.Views;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.CommunityToolkit.Effects
+{
+	static class BarStyle
+	{
+		internal static bool IsSupported()
+		{
+			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
+			{
+				Log.Warning(string.Empty, $"This functionality is not available. Minimum supported API is {BuildVersionCodes.M}");
+				return false;
+			}
+
+			return true;
+		}
```

Comment:
The SO version will not change while the app is running, so we can cache the result.

```suggestion
		static bool? isSupported = null;

		internal static bool IsSupported()
		{
			if (isSupported.HasValue)
			{
				ShowWarning();
				return isSupported.Value;
			}

			if (Build.VERSION.SdkInt < BuildVersionCodes.M)
			{
				ShowWarning();
				isSupported = false;
				return false;
			}

			isSupported = true;
			return true;

			static void ShowWarning() => Log.Warning(string.Empty, $"This functionality is not available. Minimum supported API is {BuildVersionCodes.M}");
		}
```

Reply:
That check is free since `SdkInt` is constant, so it doesn't really matter if you're checking `isSupported.HasValue` or `Build.VERSION.SdkInt < BuildVersionCodes.M`. Caching just complicates the code here.